### PR TITLE
User should see an error message when MP returns an error

### PR DESCRIPTION
**Pull request by yuki24**
finishes https://artsyproduct.atlassian.net/browse/PURCHASE-227 and https://artsyproduct.atlassian.net/browse/PURCHASE-187

u

### CODE REVIEW

File: src/lib/Components/Bidding/Screens/__tests__/ConfirmBid-tests.tsx [hunk 1]
```diff
@@ -141,7 +141,7 @@ describe("when pressing bid button", () => {
         expect(mockphysics).not.toHaveBeenCalled()
       })
 
-      it("displays an error message", () => {
+      it("displays an error message on a netwrok failure", () => {
```
Comment (sweir27) [inline src/lib/Components/Bidding/Screens/__tests__/ConfirmBid-tests.tsx]: #superminor `network`

File: src/lib/Components/Bidding/Screens/__tests__/ConfirmBid-tests.tsx [hunk 2]
```diff
@@ -162,6 +162,34 @@ describe("when pressing bid button", () => {
           })
         )
       })
+
+      it("displays an error message on a mutation failure", () => {
+        const error = {
+          message:
+            'https://stagingapi.artsy.net/api/v1/bidder?sale_id=leclere-impressionist-and-modern-art - {"error":"Invalid Sale"}',
```
Comment (sweir27) [inline src/lib/Components/Bidding/Screens/__tests__/ConfirmBid-tests.tsx]: A little confusing-- the url in this error message is from a `createBidder` mutation it looks like, not a `createBidderPosition` mutation (although they'll likely have the same error message, so this is #minor).

**Comment by sweir27** [comment]
Could you update the registration flow to have similar error handling? (Difference being instead of being able to call `presentError` we'd just be `console.log`ing at this point.

If you want to leave that out it's 👍 I'll just try to remember to put it in a subsequent PR!

File: src/lib/Components/Bidding/Screens/__tests__/ConfirmBid-tests.tsx (hunk 2 above)
Comment (yuki24) [reply src/lib/Components/Bidding/Screens/__tests__/ConfirmBid-tests.tsx]: Right, I had the same feeling. I'll update this to a more realistic error message.

File: src/lib/Components/Bidding/Screens/__tests__/ConfirmBid-tests.tsx (hunk 1 above)
Comment (yuki24) [reply src/lib/Components/Bidding/Screens/__tests__/ConfirmBid-tests.tsx]: 💥

**Comment by yuki24** [comment]
I'm marking this as WIP as there is still some work that needs to be done.

File: src/lib/Components/Bidding/Screens/Registration.tsx [hunk 3]
```diff
@@ -143,17 +155,34 @@ export class Registration extends React.Component<RegistrationProps, Registratio
...
       mutation: bidderMutation,
       variables: { input: { sale_id: this.props.sale.id } },
     })
   }
 
+  presentErrorResult(error, message) {
+    console.error(error)
+
+    this.props.navigator.push({
+      component: BidResultScreen,
```
Comment (yuki24) [inline src/lib/Components/Bidding/Screens/Registration.tsx]: We are re-using the `BidResultScreen` screen, but this doesn't seem right. We should switch that to the new `RegistrationResult` component once https://artsyproduct.atlassian.net/browse/PURCHASE-202 is done. Alternatively, we could add a separate `ErrorScreen` component that only shows an error message, but I'll leave it to whoever works on PURCHASE-202.

**Comment by yuki24** [comment]
This is good for review.

File: src/lib/Components/Bidding/Screens/Registration.tsx [hunk 4]
```diff
@@ -143,17 +155,34 @@ export class Registration extends React.Component<RegistrationProps, Registratio
...
   createBidder() {
     commitMutation(this.props.relay.environment, {
-      onCompleted: () => {
-        console.log("COMPLETED!")
-      },
-      onError: () => {
-        console.log("ERROR!")
-      },
+      onCompleted: (results, errors) =>
+        isEmpty(errors) ? console.log("COMPLETED!", results) : this.presentErrorResult(errors, msgForGraohQLError),
```
Comment (yuki24) [inline src/lib/Components/Bidding/Screens/Registration.tsx]: I left the `console.log` so we can incrementally make changes and work on https://artsyproduct.atlassian.net/browse/PURCHASE-203.

File: src/lib/Components/Bidding/Screens/Registration.tsx [hunk 5]
```diff
@@ -66,6 +69,16 @@ const bidderMutation = graphql`
   }
 `
 
+const msgForNetworkError = {
+  message_header: "An error occurred",
+  message_description_md: "Please\ncheck your internet connection\nand try again.",
+}
+
+const msgForGraohQLError = {
```
Comment (sweir27) [inline src/lib/Components/Bidding/Screens/Registration.tsx]: #minor `msgForGraolQLError` --> `msgForGraphQLError`

**Comment by sweir27** [comment]
This looks great @yuki24 ! 👏

**Pull request merged**